lib.rs

```diff
@@ -87,16 +140,24 @@ impl<D: Decorator> Format<D> {
         let r_decorator = self.decorator.decorate(record);
 
 
-        try!(self.print_msg_header(io, &r_decorator, record));
+        let mut skip_msg_comma = try!(self.print_msg_header(io, &r_decorator, record));
```

Comment:
It seems to me that `comma_need` is a better name. Also instead of having a local value here, it would be better style to have a member variable shared between compact and full code.